syntax = "proto3";

package message;

message Message {
  string room_id = 1;
  string sender = 2;
  string content = 3;
  int64 timestamp = 4;
}

message SendResponse {
  bool success = 1;
  string message = 2;
}

message HistoryRequest {
  string room_id = 1;
}

message HistoryResponse {
  repeated Message messages = 1;
}

service MessageService {
  rpc SendMessage (Message) returns (SendResponse);
  rpc GetMessages (HistoryRequest) returns (HistoryResponse);
}
